### DISCUSSION TITLE
Wait for render to complete

### DISCUSSION THREAD

**Discussion by marcuslindblom** [lit/lit#2594] [Q&A]
When I navigate in my app I first load the "base" HTML with like navigation and such. When this is in place I would like to apply the partial view content. Is it possible to wait for render to complete like `await render(...)`? Or is there any other solution to this problem?

```js
page('/', (ctx, next) => {
  render(
    html`<str-app></str-app>`,
    document.body,
    { renderBefore: document.scripts.namedItem('main') },
  )
  next();
}, () => {
  render(
    html`<h1>Hello world</h1>`,
    document.getElementById('app'),
  );
});
```

**Comment by justinfagnani** [comment]
`render()` is synchronous, so you can just run code after it.

If `<str-app>` is a LitElement, then that renders asynchronously, and you can wait for it with `await el.updateComplete`.

**Reply by marcuslindblom** [reply]
Ok, thanks!

Maybe it is ab anti pattern to do this? Should I just render all HTML on every route instead?
